### PR TITLE
csharp.md Reviewed for technical authoring.

### PR DESCRIPTION
**Pull request by gopher504**
csharp.md - made use of "voice" consistent, changed layout slightly, formalised introduction of abbreviations etc.

### CODE REVIEW

File: coding/csharp.md [hunk 1]
```diff
@@ -105,12 +105,12 @@ public class MyObject
...
 
-- Methods should aim to be around 4 to 6 lines. 8 – 10 lines but you need a really good reason.  White space is not counted, as we do like things to look nice.
-- The width of a line should not exceed 120 characters to make sure you don’t have to scroll right in the IDE.
+- Methods should aim to be around 4 to 6 lines, 8 – 10 lines with a really good reason.  White space is not counted
+- The width of a line should not exceed 120 characters to prevent scrolling right in the IDE
 
 ###### EXAMPLE
 
-You are using a lot of lines to set properties on an object. Because you’ve limited your constructor arguments. Put them in a private function.
+Because a lot of lines have been used to set properties on an object, due to limited constructor arguments, they should be put in a private function.
```
Comment (stevemchugh) [inline coding/csharp.md]: I was always taught not to start a sentence with Because

File: coding/csharp.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ Any exceptions should be agreed by the project team.
 
 ### ReSharper
 
-ReSharper must be installed and used for development in Visual Studio. You also need to ensure you have referenced the Backend Developers ReSharper .dotsettings file, which contains a standardised set of ReSharper settings for the team, in the following location:
+ReSharper must be installed and used for development in Visual Studio. You also need to ensure there is a  referenced to the Backend Developers ReSharper .dotsettings file, which contains a standardised set of ReSharper settings for the team, in the following location:
```
Comment (stevemchugh) [inline coding/csharp.md]: referenced should be reference?

File: coding/csharp.md [hunk 3]
```diff
@@ -169,15 +169,15 @@ OpenFreePopup(title, description);
...
 
 ### Block Statements
 
-- Braces should appear at the start of the line, not at the end of the previous line.
-- Always use curly braces for single line blocks, unless the code is more understandable and reads better without them.
-- One block statement per function!  Never more! This includes if’s while’s and switch statements.
-- Use of a switch statements can often be replaced with a factory or strategy design pattern to remove it. If you are not sure how to implement, ask your peers.
+- Braces should appear at the start of the line, not at the end of the previous line
+- Always use curly braces for single line blocks, unless the code is more understandable and reads better without them
+- One block statement per function, never more. This includes ifs whiles and switch statements
```
Comment (stevemchugh) [inline coding/csharp.md]: Do we need a comma between ifs and whiles?

File: coding/csharp.md [hunk 4]
```diff
@@ -219,19 +219,21 @@ Default indentation for .NET projects should be 4 spaces. ReSharper settings wil
 
 ### Using Statements
 
-Using statements should list the System.x namespaces at the top of the file, followed by other namespaces alphabetised.
+Using statements must appear at the top of the source file. Unused using statements must be deleted as they can cause compilation failures.
```
Comment (stevemchugh) [inline coding/csharp.md]: I think this has lost a bit of the intent

File: coding/csharp.md [hunk 5]
```diff
@@ -219,19 +219,21 @@ Default indentation for .NET projects should be 4 spaces. ReSharper settings wil
...
-- No NOISE comments. E.g TODO To do means you didn’t need it right now! If you don’t need it right now, you are not using test driven development. Remove it.
-- Document comments are allowed for public classes/methods only, and when the API is likely to be consumed outside the team or by the team as a compiled component e.g. as a NuGet package.
-- If you see commented out code... Delete it! It should never exist. The same applies for if you read a comment and it doesn’t make sense.
+- Comments should be set to red within the IDE
+- Comments should only be used in exceptional circumstances
+- Most comments are not needed and are prone to becoming out of date
+- If a comment is considered necessary then rewrite the code to make it more readable
+- No NOISE comments, e.g. TODO, to do means it was not needed at the time and test driven development was not being followed
+- Document comments are allowed for public classes/methods only, and when the API is likely to be consumed outside the team or by the team as a compiled component e.g. as a NuGet package
+- If commented out code if found it must be deleted
```
Comment (stevemchugh) [inline coding/csharp.md]: if to is

File: coding/csharp.md (hunk 1 above)
Comment (gopher504) [inline coding/csharp.md]: The interweb and blogosphere seem to this it's ok.

File: coding/csharp.md (hunk 2 above)
Comment (gopher504) [inline coding/csharp.md]: Typo corrected.

File: coding/csharp.md (hunk 3 above)
Comment (gopher504) [inline coding/csharp.md]: Yes, good spot.

File: coding/csharp.md (hunk 4 above)
Comment (gopher504) [inline coding/csharp.md]: System portion reinstated.

File: coding/csharp.md (hunk 5 above)
Comment (gopher504) [inline coding/csharp.md]: Corrected.

**Comment by stevemchugh** [comment]
LGTM

**Pull request merged**